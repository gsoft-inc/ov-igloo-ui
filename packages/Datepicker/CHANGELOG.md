# @igloo-ui/datepicker

## 0.0.3

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console
- Updated dependencies [6ee9c45]
  - @igloo-ui/button@0.3.7
  - @igloo-ui/dropdown@0.2.1
  - @igloo-ui/icon-button@0.2.13

## 0.0.2

### Patch Changes

- Updated dependencies [64ca515]
  - @igloo-ui/dropdown@0.2.0

## 0.0.1

### Patch Changes

- 0640409: Update @igloo-ui/icons to 1.6.0
- 0640409: Update packages

  - "react-aria": "^3.21.0",
  - "@react-aria/link": "^3.35",
  - "react-stately": "^3.19.0"

- 0640409: udpate classnames to 2.3.2
- Updated dependencies [0640409]
- Updated dependencies [0640409]
- Updated dependencies [0640409]
  - @igloo-ui/button@0.3.6
  - @igloo-ui/icon-button@0.2.12
  - @igloo-ui/dropdown@0.1.1
