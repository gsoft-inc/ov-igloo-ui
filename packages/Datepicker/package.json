{
  "name": "@igloo-ui/datepicker",
  "version": "0.6.8",
  "main": "dist/Datepicker.js",
  "module": "dist/Datepicker.js",
  "types": "dist/Datepicker.d.ts",
  "styles": "dist/datepicker.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Franck franck.gaudin@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "@igloo-ui/provider": ">=0.1.1",
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/button": "^0.8.5",
    "@igloo-ui/dropdown": "^1.8.1",
    "@igloo-ui/icon-button": "^1.2.7",
    "@igloo-ui/icons": "^1.12.2",
    "@igloo-ui/tokens": "^2.0.0",
    "@hopper-ui/tokens": "^3.5.0",
    "@internationalized/date": "^3.5.2",
    "classnames": "^2.3.2",
    "luxon": "^3.4.4",
    "react-aria": "^3.32.1",
    "react-stately": "^3.30.1"
  },
  "devDependencies": {
    "@igloo-ui/provider": "^0.1.1"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
