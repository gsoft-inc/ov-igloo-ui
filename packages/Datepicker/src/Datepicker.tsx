import * as React from 'react';
import cx from 'classnames';
import {
  getLocalTimeZone,
  now,
  parseAbsoluteToLocal,
  ZonedDateTime,
} from '@internationalized/date';

import Calendar from './components/Calendar';

import './datepicker.scss';

type Date = { utc: string; local: string };

export interface DatepickerProps {
  /** Callback function that will be called when the user types something. */
  onChange?: (date: Date) => void;
  /** Selected value for the date picker. */
  selectedDay?: string;
  /** True if the date picker should be disabled. */
  disabled?: boolean;
  /** Add a data-test tag for automated tests. */
  dataTest?: string;
}

const Datepicker: React.FunctionComponent<DatepickerProps> = (
  props: DatepickerProps
) => {
  const { selectedDay, onChange, disabled = false, dataTest } = props;

  // the calendar receives an utc date and formats it locally
  const formattedDate = selectedDay
    ? parseAbsoluteToLocal(selectedDay)
    : undefined;

  const dateTimeOfDay = now(getLocalTimeZone());

  const handleChange = (date: { year: number; month: number; day: number }) => {
    const { year, month, day } = date;
    const { hour, minute, second, millisecond, offset, timeZone } =
      dateTimeOfDay;

    const local = new ZonedDateTime(
      year,
      month,
      day,
      timeZone,
      offset,
      minute,
      hour,
      second,
      millisecond
    );

    // the calendar return an object with utc and local
    if (onChange) {
      onChange({
        utc: local.toAbsoluteString(),
        local: local.toString(),
      });
    }
  };

  const classes = cx('ids-datepicker', {
    'ids-datepicker--disabled': disabled,
  });

  return (
    <>
      <Calendar
        dataTest={dataTest}
        aria-label="Event date"
        className={classes}
        value={formattedDate}
        onChange={handleChange}
        isDisabled={disabled}
      />
    </>
  );
};

export default Datepicker;
