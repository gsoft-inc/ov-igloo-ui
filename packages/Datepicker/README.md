# Datepicker

Date pickers let users choose dates from a visual calendar thatâ€™s consistently applied wherever dates need to be selected across Officevibe.

<ReferenceLinks />

## Installation

To install `@igloo-ui/datepicker` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/datepicker
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/datepicker
```

## Usage

Then to use the component in your code just import it!

```jsx
import Datepicker from '@igloo-ui/datepicker';
const [date, setDate] = useState('2022-09-30T00:00:00Z');

const handleChange = (date) => {
  setDate(date.utc);
};

<Datepicker selectedDay={date} onChange={handleChange} />;
```
