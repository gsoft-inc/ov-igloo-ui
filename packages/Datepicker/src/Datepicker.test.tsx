/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';

import Datepicker from './Datepicker';

describe('Datepicker', () => {
  const component = shallow(<Datepicker />);
  test('It should render without errors', () => {
    const wrapper = component.find('.ids-datepicker');
    expect(wrapper.length).toBe(1);
  });

  test('It should render a snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
