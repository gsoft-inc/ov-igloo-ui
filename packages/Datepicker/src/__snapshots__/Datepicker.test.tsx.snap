// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datepicker It should render a snapshot 1`] = `ShallowWrapper {}`;
